bindings/core.ts

```diff
@@ -0,0 +1,130 @@
+import { bindSolcMethod, bindSolcMethodWithFallbackFunc } from './helpers';
+import translate from '../translate';
+import * as semver from 'semver';
+
+export function setupCore (solJson) {
+  const core = {
+    alloc: bindAlloc(solJson),
+    license: bindLicense(solJson),
+    version: bindVersion(solJson),
+    reset: bindReset(solJson)
```

Comment:
Here are some thoughts on the overall structure:
- I don't think `license` and `version` really fit here. These are exposed by the wrapper while all the other methods in `core` are just helpers used by other methods. I'd move them to a separate module (could be e.g. `compiler-info.ts` or maybe `identify.ts`?).
- I'd rename `core.ts` to `internal.ts` because otherwise it's a bit weird that compilation is not in core - that's core functionality of a compiler.
- Why not put callbacks in a separate module?

Reply:
Can do but callbacks are not bindings they are directly related to the compiling and should stay related (imo), will make this kind of change.

Reply:
@cameel this creates a lot of additional implementation detail that adds more complexity than required. Taking those out results in passing another set of values deep into the callback code which is already complex.

Reply:
ok. I thought moving them would be easy but if it does introduce that much complexity then let's leave them as is.

I'll take a closer look when I'm back next week but overall this is not a deal breaker.

Reply:
So apart from callbacks there are still the first 2 points. `core` really does not sound right to me here since the stuff in it is so peripheral. Maybe it would make more sense as `memory` (containing alloc, reset, the string helpers) and `info` (containing license, version, semver, etc.). Or something like that.

The only way it makes sense to me to call these `core` is by saying they're all equal parts of [`libsolc.h`](https://github.com/ethereum/solidity/blob/develop/libsolc/libsolc.h), which is true but still...

But anyway, at this point I just want to get this merged so that's just an optional suggestion. If you can adjust it easily, great, if not, we'll go with what it's already there anyway.

Reply:
I believe this still stands.